Fix and enable awsproviderlint AWSAT001 check: prefer resource.TestCheckResourceAttrPair() or ARN check functions

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Description

When checking ARN values in acceptance testing, it is preferred to use `resource.TestCheckResourceAttrPair()` to compare the value against another state value (typical for data sources) or use one of the other available ARN checking functions provided by the Terraform AWS Provider such as `testAccMatchResourceAttrRegionalARN`. See also: https://github.com/terraform-providers/terraform-provider-aws/blob/master/awsproviderlint/passes/AWSAT001/README.md

Due to the amount of reports and potential review complexity, any fixes should likely be done per-resource.

e.g.

```
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_acm_certificate_test.go:44:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_acm_certificate_test.go:50:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_acm_certificate_test.go:56:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_acm_certificate_test.go:62:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_acm_certificate_test.go:68:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_acm_certificate_test.go:74:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_acm_certificate_test.go:122:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_acm_certificate_test.go:128:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_acm_certificate_test.go:134:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_ecr_repository_test.go:23:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_ecs_task_definition_test.go:26:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_iam_instance_profile_test.go:23:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_iam_instance_profile_test.go:29:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_iam_policy_test.go:26:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_iam_user_test.go:26:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_instance_test.go:26:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_kms_alias_test.go:26:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_kms_alias_test.go:28:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_s3_bucket_test.go:26:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_ssm_parameter_test.go:24:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_ssm_parameter_test.go:36:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_ssm_parameter_test.go:61:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_subnet_test.go:49:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_subnet_test.go:66:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_subnet_test.go:83:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_subnet_test.go:100:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_subnet_test.go:117:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/data_source_aws_subnet_test.go:134:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_acm_certificate_test.go:67:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_acm_certificate_test.go:98:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_acm_certificate_test.go:131:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_acm_certificate_test.go:165:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_acm_certificate_test.go:230:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_acm_certificate_test.go:270:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_acm_certificate_test.go:311:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_acm_certificate_test.go:395:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_acm_certificate_test.go:429:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_acm_certificate_test.go:467:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_acm_certificate_validation_test.go:26:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_acm_certificate_validation_test.go:68:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_acm_certificate_validation_test.go:103:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_acm_certificate_validation_test.go:122:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_acm_certificate_validation_test.go:142:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_acm_certificate_validation_test.go:161:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_acm_certificate_validation_test.go:180:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_api_gateway_account_test.go:34:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_api_gateway_account_test.go:48:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_api_gateway_stage_test.go:101:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_api_gateway_stage_test.go:112:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_api_gateway_stage_test.go:122:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_api_gateway_stage_test.go:132:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_appautoscaling_target_test.go:163:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_cloud9_environment_ec2_test.go:34:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_cloud9_environment_ec2_test.go:35:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_cloud9_environment_ec2_test.go:50:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_cloud9_environment_ec2_test.go:51:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_cloud9_environment_ec2_test.go:80:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_cloud9_environment_ec2_test.go:81:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_cloud9_environment_ec2_test.go:97:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_cloud9_environment_ec2_test.go:98:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_cloudfront_distribution_test.go:335:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_cloudfront_origin_access_identity_test.go:34:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_cloudfront_origin_access_identity_test.go:67:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_cloudwatch_event_target_test.go:112:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_cloudwatch_event_target_test.go:122:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_cloudwatch_event_target_test.go:152:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_cloudwatch_event_target_test.go:184:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_codedeploy_deployment_group_test.go:41:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_codedeploy_deployment_group_test.go:74:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_codedeploy_deployment_group_test.go:127:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_codedeploy_deployment_group_test.go:162:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_codepipeline_test.go:33:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_codepipeline_test.go:75:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_codepipeline_test.go:117:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_config_aggregate_authorization_test.go:74:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_config_config_rule_test.go:57:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_config_config_rule_test.go:95:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_config_configuration_aggregator_test.go:109:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_config_configuration_recorder_test.go:85:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_config_configuration_recorder_test.go:112:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_config_delivery_channel_test.go:119:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_db_cluster_snapshot_test.go:31:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_db_event_subscription_test.go:31:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_db_parameter_group_test.go:107:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_db_parameter_group_test.go:145:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_db_security_group_test.go:36:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_db_subnet_group_test.go:91:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_docdb_cluster_parameter_group_test.go:31:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_dynamodb_global_table_test.go:42:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_dynamodb_global_table_test.go:71:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_eks_cluster_test.go:84:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_eks_cluster_test.go:93:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_elastic_transcoder_pipeline_test.go:61:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_gamelift_fleet_test.go:276:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_gamelift_fleet_test.go:295:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_gamelift_fleet_test.go:358:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_gamelift_fleet_test.go:398:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_glue_job_test.go:78:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_instance_test.go:289:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_iot_role_alias_test.go:65:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_kinesis_firehose_delivery_stream_test.go:735:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_lambda_alias_test.go:38:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_lambda_event_source_mapping_test.go:57:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_lambda_event_source_mapping_test.go:154:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_lambda_event_source_mapping_test.go:185:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_lambda_function_test.go:86:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_lambda_function_test.go:408:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_lambda_function_test.go:462:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_lambda_permission_test.go:451:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_mq_broker_test.go:275:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_mq_broker_test.go:352:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_mq_broker_test.go:461:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_neptune_cluster_instance_test.go:30:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_neptune_cluster_instance_test.go:163:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_neptune_cluster_parameter_group_test.go:31:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_neptune_cluster_snapshot_test.go:31:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_neptune_cluster_test.go:217:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_neptune_event_subscription_test.go:30:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_neptune_parameter_group_test.go:30:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_organizations_policy_test.go:31:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_ram_resource_share_accepter_test.go:34:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_ram_resource_share_accepter_test.go:35:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_rds_cluster_parameter_group_test.go:101:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_rds_cluster_parameter_group_test.go:194:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_rds_cluster_test.go:117:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_s3_bucket_test.go:183:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_secretsmanager_secret_test.go:81:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_secretsmanager_secret_test.go:116:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_secretsmanager_secret_test.go:249:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_secretsmanager_secret_version_test.go:33:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_secretsmanager_secret_version_test.go:64:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_security_group_test.go:694:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_sns_platform_application_test.go:186:8: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_sns_sms_preferences_test.go:101:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_sns_topic_test.go:220:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_sns_topic_test.go:222:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_sns_topic_test.go:223:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_sns_topic_test.go:225:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_sns_topic_test.go:226:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_sns_topic_test.go:228:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_sns_topic_test.go:229:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_sns_topic_test.go:231:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_storagegateway_cache_test.go:86:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_storagegateway_cache_test.go:113:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_storagegateway_nfs_file_share_test.go:34:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_storagegateway_nfs_file_share_test.go:38:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_storagegateway_nfs_file_share_test.go:43:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_storagegateway_nfs_file_share_test.go:254:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_storagegateway_nfs_file_share_test.go:262:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_storagegateway_smb_file_share_test.go:33:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_storagegateway_smb_file_share_test.go:38:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_storagegateway_smb_file_share_test.go:42:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_storagegateway_smb_file_share_test.go:73:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_storagegateway_smb_file_share_test.go:78:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_storagegateway_smb_file_share_test.go:82:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_storagegateway_smb_file_share_test.go:290:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_storagegateway_smb_file_share_test.go:298:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_storagegateway_upload_buffer_test.go:86:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_storagegateway_working_storage_test.go:86:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_subnet_test.go:156:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_waf_ipset_test.go:39:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
/Users/bflad/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_wafregional_ipset_test.go:40:6: AWSAT001: prefer resource.TestCheckResourceAttrPair() or ARN check functions (e.g. testAccMatchResourceAttrRegionalARN)
```

### Definition of Done

- Fix AWSAT001 reports
- In `GNUmakefile`, add `-AWSAT001` to `awsproviderlint` command under lint target and have TravisCI testing pass

--- a/GNUmakefile
+++ b/GNUmakefile
@@ -67,6 +67,7 @@ lint:
 		-AT006 \
 		-AT007 \
 		-AT008 \
+        -AWSAT001 \
 		-AWSR001 \
 		-AWSR002 \
 		-R002 \

--- a/GNUmakefile
+++ b/GNUmakefile
@@ -67,7 +67,7 @@ lint:
 		-AT006 \
 		-AT007 \
 		-AT008 \
-        -AWSAT001 \
+		-AWSAT001 \
 		-AWSR001 \
 		-AWSR002 \
 		-R002 \

